package/MDAnalysis/visualization/streamlines.py

```diff
@@ -149,11 +149,13 @@ def split_grid(grid, num_cores):
     # be distributed over the available cores on the system:
     list_square_vertex_arrays_per_core = np.array_split(list_all_squares_in_grid, num_cores)
     list_parent_index_values = np.array_split(list_parent_index_values, num_cores)
-    return [list_square_vertex_arrays_per_core, list_parent_index_values, current_row, current_column]
+    return (list_square_vertex_arrays_per_core, list_parent_index_values, current_row, current_column)
 
 
-def per_core_work(topology_file_path, trajectory_file_path, list_square_vertex_arrays_this_core, MDA_selection,
-                  start_frame, end_frame, reconstruction_index_list, maximum_delta_magnitude):
+def per_core_work(topology_file_path: str, trajectory_file_path: str,
+                  list_square_vertex_arrays_this_core: List[List[float]],
+                  MDA_selection: str, start_frame: int, end_frame: int, reconstruction_index_list: Sequence[int],
+                  maximum_delta_magnitude: float) -> None:
```

Comment:
```suggestion
def per_core_work(
    topology_file_path: str, trajectory_file_path: str,
    list_square_vertex_arrays_this_core: List[List[float]],
    MDA_selection: str, start_frame: int, end_frame: int,
    reconstruction_index_list: Sequence[int],
    maximum_delta_magnitude: float,
) -> List[Tuple[int, List[float]]]:
```

I think I get the return type correct here, but the code is a bit convoluted. In any case, this function *does* return something as I wrote you in an email.